File: app/src/main/java/com/jiang/thinkindler/ui/MainActivity.java
package com.jiang.thinkindler.ui;

import android.os.Bundle;
import android.support.design.widget.NavigationView;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.jiang.common.base.CommonActivity;
import com.jiang.common.base.CommonFragment;
import com.jiang.douban.ui.main.DoubanMainFragment;
import com.jiang.meizi.MeiziMainFragment;
import com.jiang.thinkindler.R;

import butterknife.BindView;

public class MainActivity extends CommonActivity {

    private DoubanMainFragment doubanFragment;
    private MeiziMainFragment meiziFragment;

    private CommonFragment[] mFragments;
    @BindView(R.id.nav_main)
    NavigationView navigationView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());

        init();
    }

    public int getLayoutId() {
        return R.layout.act_main;
    }

    protected void init() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open,
                R.string.navigation_drawer_close);
        if (drawer != null) {
            drawer.addDrawerListener(toggle);
        }
        toggle.syncState();

        toolbar.inflateMenu(R.menu.menu_clear);

        mFragments = new CommonFragment[2];
        doubanFragment = (DoubanMainFragment) getSupportFragmentManager()
                .findFragmentById(R.id.frag_main_douban);
        mFragments[0] = doubanFragment;
        meiziFragment = (MeiziMainFragment) getSupportFragmentManager()
                .findFragmentById(R.id.frag_main_meizi);
        mFragments[1] = meiziFragment;

        displayFragmentByIndex(0);

        navigationView.setNavigationItemSelectedListener(onNavigationItemSelectedListener);

    }


    NavigationView.OnNavigationItemSelectedListener onNavigationItemSelectedListener
            = item -> {
        switch (item.getItemId()) {
            case R.id.nav_douban:
                displayFragmentByIndex(0);
                break;
            case R.id.nav_meizi:
                displayFragmentByIndex(1);
                break;
            default:
                break;
        }
        return true;
    };

    private void displayFragmentByIndex(int index) {
        // 通过这个底部容器Item的index能够获取到对应的Fragment，需要将所有的Fragment对号放好（使用集合）
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        for (int i = 0; i < mFragments.length; i++) {
            if (i == index) {
                ft.show(mFragments[i]);
            } else {
                ft.hide(mFragments[i]);
            }
        }
        ft.commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case R.id.action_clear:
                showShortToast(R.string.action_clear);
                break;
            default:
                break;
        }
        return true;
    }

}

File: app/src/main/java/com/jiang/thinkindler/ui/MainActivity.kt
package com.jiang.thinkindler.ui

import android.support.design.widget.NavigationView
import android.support.v4.app.Fragment
import android.support.v4.util.ArrayMap
import android.support.v4.view.GravityCompat
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import com.alibaba.android.arouter.launcher.ARouter
import com.jiang.thinkindler.R
import com.jiang.thinkindler.base.BaseActivity
import org.jetbrains.anko.find


class MainActivity : BaseActivity() {

    private val navigationView by lazy { find<NavigationView>(R.id.nav_main) }

    private val toolbar by lazy { find<Toolbar>(R.id.toolbar) }

    private val drawer by lazy { find<DrawerLayout>(R.id.drawer_layout) }

    private var fragments = ArrayMap<String, Fragment>()

    override fun getLayoutId(): Int {
        return R.layout.act_main
    }

    override fun init() {

        setSupportActionBar(toolbar)

        val toggle = ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open,
                R.string.navigation_drawer_close)
        drawer.addDrawerListener(toggle)
        toggle.syncState()

        toolbar.inflateMenu(R.menu.menu_clear)

        navigationView.setNavigationItemSelectedListener({
            when (it.itemId) {
                R.id.nav_douban -> displayFragmentByURL("/douban/main")
                R.id.nav_meizi -> displayFragmentByURL("/meizi/main")
                R.id.nav_bilibili -> displayFragmentByURL("/bilibili/main")
                else -> {
                }
            }
            drawer.closeDrawer(GravityCompat.START)
            true
        })

        //默认首页
        navigationView.post({
            navigationView.menu.findItem(R.id.nav_douban)?.isChecked = true
            displayFragmentByURL("/douban/main")
        })

    }

    override fun initInjector() {

    }

    private fun displayFragmentByURL(url: String) {

        val transaction = supportFragmentManager.beginTransaction()
        if (fragments[url] == null) {
            val fragment = (ARouter.getInstance().build(url).navigation() as Fragment?) ?: return
            fragments.put(url, fragment)
            transaction.add(R.id.ly_main_container, fragments[url])
        }

        for (entry in fragments.entries) {
            if (entry.key == url) {
                transaction.show(entry.value)
            } else {
                transaction.hide(entry.value)
            }
        }

        transaction.commit()

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        when (id) {
            R.id.action_clear -> showShortToast(R.string.action_clear)
            else -> {
            }
        }
        return true
    }

    override fun onBackPressed() {
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }

    }

}


